src/benchmark.rs

```diff
@@ -18,16 +18,24 @@ impl Miner {
             "Benchmarking. This will take {} sec...",
             TEST_DURATION
         ));
-        let handles: Vec<_> = (0..args.threads)
+        let core_ids = core_affinity::get_core_ids().unwrap();
+        let handles: Vec<_> = core_ids
+            .into_iter()
             .map(|i| {
                 std::thread::spawn({
+                    let mut memory = equix::SolverMemory::new();
                     move || {
+                        // Return if core should not be used
+                        if (i.id as u64).ge(&args.cores) {
+                            return 0;
+                        }
+                        let _ = core_affinity::set_for_current(i);
                         let timer = Instant::now();
-                        let first_nonce = u64::MAX.saturating_div(args.threads).saturating_mul(i);
+                        let first_nonce = u64::MAX.saturating_div(args.cores).saturating_mul(i.id as u64);
                         let mut nonce = first_nonce;
                         loop {
                             // Create hash
-                            let _hx = drillx::hash(&challenge, &nonce.to_le_bytes());
+                            let _hx = drillx::hash_with_memory(&mut memory,&challenge, &nonce.to_le_bytes());
```

Comment:
while you are at it, only count the hashes, instead of nonces. This call may return an error, and it is not handled here.

Reply:
Thank you. Here I just change the original code from `drillx::hash` to `drillx::hash_with_memory` to aligh with the `mine` function.
`drillx::hash` and `drillx::hash_with_memory` have same return `Result<Hash, DrillxError>`

It's possible to optimise it.